fix linting errors from PEP484

nox linting fails due to changes in accordance with PEP 484. Here's the key messages:
- `planet/cli/session.py:9: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True`
- 'Incompatible default for argument "state" (default has type "None", argument has type "str")  [assignment]'
- 'Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase'

And here's the full list of failures:
```
planet/reporting.py:60: error: Incompatible default for argument "order_id" (default has type "None", argument has type "str")  [assignment]
planet/reporting.py:60: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/reporting.py:60: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/reporting.py:61: error: Incompatible default for argument "state" (default has type "None", argument has type "str")  [assignment]
planet/reporting.py:61: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/reporting.py:61: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/reporting.py:91: error: Incompatible default for argument "state" (default has type "None", argument has type "str")  [assignment]
planet/reporting.py:91: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/reporting.py:91: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/reporting.py:91: error: Incompatible default for argument "order_id" (default has type "None", argument has type "str")  [assignment]
planet/models.py:64: error: Incompatible return value type (got "Headers", expected "List[Any]")  [return-value]
planet/models.py:68: error: Incompatible return value type (got "URL", expected "str")  [return-value]
planet/order_request.py:28: error: Incompatible default for argument "delivery" (default has type "None", argument has type "Dict[Any, Any]")  [assignment]
planet/order_request.py:28: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:28: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:29: error: Incompatible default for argument "notifications" (default has type "None", argument has type "Dict[Any, Any]")  [assignment]
planet/order_request.py:29: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:29: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:30: error: Incompatible default for argument "order_type" (default has type "None", argument has type "str")  [assignment]
planet/order_request.py:30: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:30: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:31: error: Incompatible default for argument "tools" (default has type "None", argument has type "List[Dict[Any, Any]]")  [assignment]
planet/order_request.py:31: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:31: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:32: error: Incompatible default for argument "stac" (default has type "None", argument has type "Dict[Any, Any]")  [assignment]
planet/order_request.py:32: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:32: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:99: error: Incompatible default for argument "fallback_bundle" (default has type "None", argument has type "str")  [assignment]
planet/order_request.py:99: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:99: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:134: error: Incompatible default for argument "email" (default has type "None", argument has type "bool")  [assignment]
planet/order_request.py:134: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:134: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:135: error: Incompatible default for argument "webhook_url" (default has type "None", argument has type "str")  [assignment]
planet/order_request.py:135: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:135: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:136: error: Incompatible default for argument "webhook_per_order" (default has type "None", argument has type "bool")  [assignment]
planet/order_request.py:136: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:136: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:148: error: Incompatible default for argument "archive_type" (default has type "None", argument has type "str")  [assignment]
planet/order_request.py:148: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:148: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:150: error: Incompatible default for argument "archive_filename" (default has type "None", argument has type "str")  [assignment]
planet/order_request.py:150: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:150: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:151: error: Incompatible default for argument "cloud_config" (default has type "None", argument has type "Dict[Any, Any]")  [assignment]
planet/order_request.py:151: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:151: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:199: error: Incompatible default for argument "path_prefix" (default has type "None", argument has type "str")  [assignment]
planet/order_request.py:199: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:199: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:227: error: Incompatible default for argument "storage_endpoint_suffix" (default has type "None", argument has type "str")  [assignment]
planet/order_request.py:227: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:227: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:228: error: Incompatible default for argument "path_prefix" (default has type "None", argument has type "str")  [assignment]
planet/order_request.py:228: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:228: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:258: error: Incompatible default for argument "path_prefix" (default has type "None", argument has type "str")  [assignment]
planet/order_request.py:258: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:258: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:371: error: Incompatible default for argument "resolution" (default has type "None", argument has type "float")  [assignment]
planet/order_request.py:371: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:371: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:372: error: Incompatible default for argument "kernel" (default has type "None", argument has type "str")  [assignment]
planet/order_request.py:372: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:372: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:392: error: Incompatible default for argument "origin_x" (default has type "None", argument has type "float")  [assignment]
planet/order_request.py:392: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:392: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:393: error: Incompatible default for argument "origin_y" (default has type "None", argument has type "float")  [assignment]
planet/order_request.py:393: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:393: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:394: error: Incompatible default for argument "pixel_size" (default has type "None", argument has type "float")  [assignment]
planet/order_request.py:394: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:394: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:395: error: Incompatible default for argument "name_template" (default has type "None", argument has type "str")  [assignment]
planet/order_request.py:395: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:395: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:396: error: Incompatible default for argument "conformal_x_scaling" (default has type "None", argument has type "bool")  [assignment]
planet/order_request.py:396: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:396: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/order_request.py:417: error: Incompatible default for argument "scale_factor" (default has type "None", argument has type "int")  [assignment]
planet/order_request.py:417: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/order_request.py:417: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/auth.py:55: error: Incompatible default for argument "filename" (default has type "None", argument has type "Union[str, Path]")  [assignment]
planet/auth.py:55: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/auth.py:55: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/auth.py:80: error: Incompatible default for argument "variable_name" (default has type "None", argument has type "str")  [assignment]
planet/auth.py:80: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/auth.py:80: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/auth.py:102: error: Incompatible default for argument "base_url" (default has type "None", argument has type "str")  [assignment]
planet/auth.py:102: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/auth.py:102: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/auth.py:136: error: Incompatible default for argument "filename" (default has type "None", argument has type "Union[str, Path]")  [assignment]
planet/auth.py:136: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/auth.py:136: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/auth.py:149: error: Incompatible default for argument "base_url" (default has type "None", argument has type "str")  [assignment]
planet/auth.py:149: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/auth.py:149: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/http.py:230: error: Incompatible default for argument "auth" (default has type "None", argument has type "Auth")  [assignment]
planet/http.py:230: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/http.py:230: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/http.py:358: error: Invalid type comment or annotation  [valid-type]
planet/http.py:358: note: Suggestion: use Union[...] instead of Union(...)
planet/http.py:359: error: Incompatible default for argument "json" (default has type "None", argument has type "Dict[Any, Any]")  [assignment]
planet/http.py:359: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/http.py:359: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/http.py:360: error: Incompatible default for argument "params" (default has type "None", argument has type "Dict[Any, Any]")  [assignment]
planet/http.py:360: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/http.py:360: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/http.py:397: error: Incompatible return value type (got "Response", expected "Request")  [return-value]
planet/http.py:399: error: The return type of an async generator function should be "AsyncGenerator" or one of its supertypes  [misc]
planet/http.py:399: error: Argument 1 to "asynccontextmanager" has incompatible type "Callable[[Session, str, str], StreamingResponse]"; expected "Callable[[Session, str, str], AsyncIterator[<nothing>]]"  [arg-type]
planet/clients/orders.py:87: error: Incompatible default for argument "base_url" (default has type "None", argument has type "str")  [assignment]
planet/clients/orders.py:87: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/clients/orders.py:87: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/clients/orders.py:171: error: "OrdersClient" has no attribute "_request"  [attr-defined]
planet/clients/orders.py:172: error: "OrdersClient" has no attribute "_do_request"  [attr-defined]
planet/clients/orders.py:211: error: Incompatible default for argument "order_ids" (default has type "None", argument has type "List[str]")  [assignment]
planet/clients/orders.py:211: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/clients/orders.py:211: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/clients/orders.py:251: error: Incompatible default for argument "filename" (default has type "None", argument has type "str")  [assignment]
planet/clients/orders.py:251: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/clients/orders.py:251: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/clients/orders.py:271: error: Need type annotation for "resp"  [var-annotated]
planet/clients/orders.py:285: error: Incompatible default for argument "checksum" (default has type "None", argument has type "str")  [assignment]
planet/clients/orders.py:285: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/clients/orders.py:285: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/clients/orders.py:399: error: Incompatible default for argument "state" (default has type "None", argument has type "str")  [assignment]
planet/clients/orders.py:399: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/clients/orders.py:399: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/clients/orders.py:402: error: Incompatible default for argument "callback" (default has type "None", argument has type "Callable[[str], None]")  [assignment]
planet/clients/orders.py:402: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/clients/orders.py:402: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/clients/orders.py:463: error: Function "Callable[[str], None]" could always be true in boolean context  [truthy-function]
planet/clients/orders.py:483: error: Incompatible default for argument "state" (default has type "None", argument has type "str")  [assignment]
planet/clients/orders.py:483: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/clients/orders.py:483: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/clients/data.py:77: error: Incompatible default for argument "base_url" (default has type "None", argument has type "str")  [assignment]
planet/clients/data.py:77: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/clients/data.py:77: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/clients/data.py:105: error: Missing positional argument "url" in call to "request" of "Session"  [call-arg]
planet/clients/data.py:105: error: Argument 1 to "request" of "Session" has incompatible type "Request"; expected "str"  [arg-type]
planet/clients/data.py:110: error: Incompatible default for argument "name" (default has type "None", argument has type "str")  [assignment]
planet/clients/data.py:110: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/clients/data.py:110: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/clients/data.py:111: error: Incompatible default for argument "sort" (default has type "None", argument has type "str")  [assignment]
planet/clients/data.py:111: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/clients/data.py:111: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/clients/data.py:506: error: Incompatible return value type (got "Dict[Any, Any]", expected "List[Dict[Any, Any]]")  [return-value]
planet/clients/data.py:530: error: No overload variant of "__getitem__" of "list" matches argument type "str"  [call-overload]
planet/clients/data.py:530: note: Possible overload variants:
planet/clients/data.py:530: note:     def __getitem__(self, SupportsIndex, /) -> Dict[Any, Any]
planet/clients/data.py:530: note:     def __getitem__(self, slice, /) -> List[Dict[Any, Any]]
planet/clients/data.py:532: error: "List[Dict[Any, Any]]" has no attribute "keys"  [attr-defined]
planet/clients/data.py:572: error: Incompatible default for argument "callback" (default has type "None", argument has type "Callable[[str], None]")  [assignment]
planet/clients/data.py:572: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/clients/data.py:572: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/clients/data.py:607: error: Function "Callable[[str], None]" could always be true in boolean context  [truthy-function]
planet/clients/data.py:633: error: Incompatible return value type (got "Dict[Any, Any]", expected "str")  [return-value]
planet/clients/data.py:637: error: Incompatible default for argument "filename" (default has type "None", argument has type "str")  [assignment]
planet/clients/data.py:637: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/clients/data.py:637: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/clients/data.py:673: error: Missing positional argument "url" in call to "stream" of "Session"  [call-arg]
planet/clients/data.py:673: error: Need type annotation for "resp"  [var-annotated]
planet/clients/data.py:680: error: Incompatible return value type (got "Path", expected "str")  [return-value]
planet/data_filter.py:79: error: Incompatible default for argument "callback" (default has type "None", argument has type "Callable[..., Any]")  [assignment]
planet/data_filter.py:79: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/data_filter.py:79: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/data_filter.py:98: error: Function "Callable[..., Any]" could always be true in boolean context  [truthy-function]
planet/data_filter.py:112: error: Incompatible default for argument "gt" (default has type "None", argument has type "datetime")  [assignment]
planet/data_filter.py:112: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/data_filter.py:112: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/data_filter.py:113: error: Incompatible default for argument "lt" (default has type "None", argument has type "datetime")  [assignment]
planet/data_filter.py:113: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/data_filter.py:113: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/data_filter.py:114: error: Incompatible default for argument "gte" (default has type "None", argument has type "datetime")  [assignment]
planet/data_filter.py:114: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/data_filter.py:114: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/data_filter.py:115: error: Incompatible default for argument "lte" (default has type "None", argument has type "datetime")  [assignment]
planet/data_filter.py:115: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/data_filter.py:115: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/data_filter.py:154: error: Incompatible default for argument "gt" (default has type "None", argument has type "float")  [assignment]
planet/data_filter.py:154: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/data_filter.py:154: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/data_filter.py:155: error: Incompatible default for argument "lt" (default has type "None", argument has type "float")  [assignment]
planet/data_filter.py:155: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/data_filter.py:155: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/data_filter.py:156: error: Incompatible default for argument "gte" (default has type "None", argument has type "float")  [assignment]
planet/data_filter.py:156: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/data_filter.py:156: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/data_filter.py:157: error: Incompatible default for argument "lte" (default has type "None", argument has type "float")  [assignment]
planet/data_filter.py:157: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/data_filter.py:157: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/data_filter.py:182: error: Incompatible default for argument "gt" (default has type "None", argument has type "float")  [assignment]
planet/data_filter.py:182: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/data_filter.py:182: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/data_filter.py:183: error: Incompatible default for argument "lt" (default has type "None", argument has type "float")  [assignment]
planet/data_filter.py:183: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/data_filter.py:183: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/data_filter.py:184: error: Incompatible default for argument "gte" (default has type "None", argument has type "float")  [assignment]
planet/data_filter.py:184: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/data_filter.py:184: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/data_filter.py:185: error: Incompatible default for argument "lte" (default has type "None", argument has type "float")  [assignment]
planet/data_filter.py:185: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/data_filter.py:185: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/clients/subscriptions.py:66: error: Too many arguments for "Request"  [call-arg]
planet/clients/subscriptions.py:66: error: Unexpected keyword argument "params" for "Request"  [call-arg]
.nox/analyze/lib/python3.7/site-packages/mypy/typeshed/stdlib/builtins.pyi:90: note: "Request" defined here
planet/clients/subscriptions.py:69: error: Argument 1 to "_SubscriptionsPager" has incompatible type "Request"; expected "Dict[Any, Any]"  [arg-type]
planet/clients/subscriptions.py:95: error: Too many arguments for "Request"  [call-arg]
planet/clients/subscriptions.py:95: error: Unexpected keyword argument "method" for "Request"  [call-arg]
.nox/analyze/lib/python3.7/site-packages/mypy/typeshed/stdlib/builtins.pyi:90: note: "Request" defined here
planet/clients/subscriptions.py:95: error: Unexpected keyword argument "json" for "Request"  [call-arg]
.nox/analyze/lib/python3.7/site-packages/mypy/typeshed/stdlib/builtins.pyi:90: note: "Request" defined here
planet/clients/subscriptions.py:98: error: Missing positional argument "url" in call to "request" of "Session"  [call-arg]
planet/clients/subscriptions.py:98: error: Argument 1 to "request" of "Session" has incompatible type "Request"; expected "str"  [arg-type]
planet/clients/subscriptions.py:124: error: Too many arguments for "Request"  [call-arg]
planet/clients/subscriptions.py:124: error: Unexpected keyword argument "method" for "Request"  [call-arg]
.nox/analyze/lib/python3.7/site-packages/mypy/typeshed/stdlib/builtins.pyi:90: note: "Request" defined here
planet/clients/subscriptions.py:127: error: Missing positional argument "url" in call to "request" of "Session"  [call-arg]
planet/clients/subscriptions.py:127: error: Argument 1 to "request" of "Session" has incompatible type "Request"; expected "str"  [arg-type]
planet/clients/subscriptions.py:151: error: Too many arguments for "Request"  [call-arg]
planet/clients/subscriptions.py:151: error: Unexpected keyword argument "method" for "Request"  [call-arg]
.nox/analyze/lib/python3.7/site-packages/mypy/typeshed/stdlib/builtins.pyi:90: note: "Request" defined here
planet/clients/subscriptions.py:151: error: Unexpected keyword argument "json" for "Request"  [call-arg]
.nox/analyze/lib/python3.7/site-packages/mypy/typeshed/stdlib/builtins.pyi:90: note: "Request" defined here
planet/clients/subscriptions.py:154: error: Missing positional argument "url" in call to "request" of "Session"  [call-arg]
planet/clients/subscriptions.py:154: error: Argument 1 to "request" of "Session" has incompatible type "Request"; expected "str"  [arg-type]
planet/clients/subscriptions.py:179: error: Too many arguments for "Request"  [call-arg]
planet/clients/subscriptions.py:179: error: Unexpected keyword argument "method" for "Request"  [call-arg]
.nox/analyze/lib/python3.7/site-packages/mypy/typeshed/stdlib/builtins.pyi:90: note: "Request" defined here
planet/clients/subscriptions.py:182: error: Missing positional argument "url" in call to "request" of "Session"  [call-arg]
planet/clients/subscriptions.py:182: error: Argument 1 to "request" of "Session" has incompatible type "Request"; expected "str"  [arg-type]
planet/clients/subscriptions.py:228: error: Too many arguments for "Request"  [call-arg]
planet/clients/subscriptions.py:228: error: Unexpected keyword argument "params" for "Request"  [call-arg]
.nox/analyze/lib/python3.7/site-packages/mypy/typeshed/stdlib/builtins.pyi:90: note: "Request" defined here
planet/clients/subscriptions.py:231: error: Argument 1 to "_ResultsPager" has incompatible type "Request"; expected "Dict[Any, Any]"  [arg-type]
planet/cli/session.py:9: error: Incompatible default for argument "auth" (default has type "None", argument has type "Auth")  [assignment]
planet/cli/session.py:9: note: PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
planet/cli/session.py:9: note: Use https://github.com/hauntsaninja/no_implicit_optional to automatically upgrade your codebase
planet/cli/data.py:77: error: Argument 1 to "BadParameter" has incompatible type "SpecificationException"; expected "str"  [arg-type]
Found 108 errors in 11 files (checked 29 source files)
```

--- a/planet/auth.py
+++ b/planet/auth.py
@@ -27,6 +27,7 @@ import jwt
 from . import http, models
 from .constants import PLANET_BASE_URL, SECRET_FILE_PATH
 from .exceptions import AuthException
+from typing import Optional
 
 LOGGER = logging.getLogger(__name__)
 
@@ -52,7 +53,7 @@ class Auth(metaclass=abc.ABCMeta):
 
     @staticmethod
     def from_file(
-            filename: typing.Union[str, pathlib.Path] = None) -> AuthType:
+            filename: Optional[typing.Union[str, pathlib.Path]] = None) -> AuthType:
         '''Create authentication from secret file.
 
         The secret file is named `.planet.json` and is stored in the user
@@ -77,7 +78,7 @@ class Auth(metaclass=abc.ABCMeta):
         return auth
 
     @staticmethod
-    def from_env(variable_name: str = None) -> AuthType:
+    def from_env(variable_name: Optional[str] = None) -> AuthType:
         '''Create authentication from environment variable.
 
         Reads the `PL_API_KEY` environment variable
@@ -99,7 +100,7 @@ class Auth(metaclass=abc.ABCMeta):
     @staticmethod
     def from_login(email: str,
                    password: str,
-                   base_url: str = None) -> AuthType:
+                   base_url: Optional[str] = None) -> AuthType:
         '''Create authentication from login email and password.
 
         Note: To keep your password secure, the use of `getpass` is
@@ -133,7 +134,7 @@ class Auth(metaclass=abc.ABCMeta):
     def to_dict(self) -> dict:
         pass
 
-    def write(self, filename: typing.Union[str, pathlib.Path] = None):
+    def write(self, filename: Optional[typing.Union[str, pathlib.Path]] = None):
         '''Write authentication information.
 
         Parameters:
@@ -146,7 +147,7 @@ class Auth(metaclass=abc.ABCMeta):
 
 class AuthClient:
 
-    def __init__(self, base_url: str = None):
+    def __init__(self, base_url: Optional[str] = None):
         """
         Parameters:
             base_url: The base URL to use. Defaults to production

--- a/planet/auth.py
+++ b/planet/auth.py
@@ -53,7 +53,8 @@ class Auth(metaclass=abc.ABCMeta):
 
     @staticmethod
     def from_file(
-            filename: Optional[typing.Union[str, pathlib.Path]] = None) -> AuthType:
+        filename: Optional[typing.Union[str,
+                                        pathlib.Path]] = None) -> AuthType:
         '''Create authentication from secret file.
 
         The secret file is named `.planet.json` and is stored in the user
@@ -134,7 +135,8 @@ class Auth(metaclass=abc.ABCMeta):
     def to_dict(self) -> dict:
         pass
 
-    def write(self, filename: Optional[typing.Union[str, pathlib.Path]] = None):
+    def write(self,
+              filename: Optional[typing.Union[str, pathlib.Path]] = None):
         '''Write authentication information.
 
         Parameters:

--- a/planet/cli/data.py
+++ b/planet/cli/data.py
@@ -74,7 +74,7 @@ def check_item_types(ctx, param, item_types) -> Optional[List[dict]]:
             validate_item_type(item_type)
         return item_types
     except SpecificationException as e:
-        raise click.BadParameter(e)
+        raise click.BadParameter(str(e))
 
 
 def date_range_to_filter(ctx, param, values) -> Optional[List[dict]]:

--- a/planet/cli/session.py
+++ b/planet/cli/session.py
@@ -2,10 +2,11 @@
 
 from planet.auth import AuthType
 from planet.http import Session
+from typing import Optional
 
 
 class CliSession(Session):
 
-    def __init__(self, auth: AuthType = None):
+    def __init__(self, auth: Optional[AuthType] = None):
         super().__init__(auth)
         self._client.headers.update({'X-Planet-App': 'python-cli'})

--- a/planet/clients/data.py
+++ b/planet/clients/data.py
@@ -17,13 +17,12 @@ import hashlib
 import logging
 from pathlib import Path
 import time
-import typing
+from typing import Any, AsyncIterator, Callable, Dict, List, Optional
 
 from .. import exceptions
 from ..constants import PLANET_BASE_URL
 from ..http import Session
 from ..models import Paged, Request, Response, StreamingBody
-from typing import Optional
 
 BASE_URL = f'{PLANET_BASE_URL}/data/v1/'
 SEARCHES_PATH = '/searches'
@@ -106,11 +105,11 @@ class DataClient:
         return await self._session.request(request)
 
     async def search(self,
-                     item_types: typing.List[str],
+                     item_types: List[str],
                      search_filter: dict,
                      name: Optional[str] = None,
                      sort: Optional[str] = None,
-                     limit: int = 100) -> typing.AsyncIterator[dict]:
+                     limit: int = 100) -> AsyncIterator[dict]:
         """Execute a quick search.
 
         Quick searches are saved for a short period of time (~month). The
@@ -194,7 +193,7 @@ class DataClient:
 
     async def create_search(self,
                             name: str,
-                            item_types: typing.List[str],
+                            item_types: List[str],
                             search_filter: dict,
                             enable_email: bool = False) -> dict:
         """Create a new saved structured item search.
@@ -244,7 +243,7 @@ class DataClient:
     async def update_search(self,
                             search_id: str,
                             name: str,
-                            item_types: typing.List[str],
+                            item_types: List[str],
                             search_filter: dict,
                             enable_email: bool = False) -> dict:
         """Update an existing saved search.
@@ -275,7 +274,7 @@ class DataClient:
     async def list_searches(self,
                             sort: str = 'created desc',
                             search_type: str = 'any',
-                            limit: int = 100) -> typing.AsyncIterator[dict]:
+                            limit: int = 100) -> AsyncIterator[dict]:
         """List all saved searches available to the authenticated user.
 
         NOTE: the term 'saved' is overloaded here. We want to list saved
@@ -344,7 +343,7 @@ class DataClient:
 
     async def run_search(self,
                          search_id: str,
-                         limit: int = 100) -> typing.AsyncIterator[dict]:
+                         limit: int = 100) -> AsyncIterator[dict]:
         """Execute a saved search.
 
         Parameters:
@@ -364,7 +363,7 @@ class DataClient:
         return Items(request, self._do_request, limit=limit)
 
     async def get_stats(self,
-                        item_types: typing.List[str],
+                        item_types: List[str],
                         search_filter: dict,
                         interval: str) -> dict:
         """Get item search statistics.
@@ -399,7 +398,7 @@ class DataClient:
         response = await self._do_request(request)
         return response.json()
 
-    async def list_asset_types(self) -> typing.List[dict]:
+    async def list_asset_types(self) -> List[dict]:
         """List all asset types available to the authenticated user.
 
         Returns:
@@ -428,7 +427,7 @@ class DataClient:
         """
         raise NotImplementedError
 
-    async def list_item_types(self) -> typing.List[dict]:
+    async def list_item_types(self) -> List[dict]:
         """List all item types available to the authenticated user.
 
         Returns:
@@ -482,8 +481,7 @@ class DataClient:
         """
         raise NotImplementedError
 
-    async def list_item_assets(self, item_type_id: str,
-                               item_id: str) -> typing.List[dict]:
+    async def list_item_assets(self, item_type_id: str, item_id: str) -> dict:
         """List all assets available for an item.
 
         An asset describes a product that can be derived from an item's source
@@ -566,11 +564,13 @@ class DataClient:
 
         return
 
-    async def wait_asset(self,
-                         asset: dict,
-                         delay: int = WAIT_DELAY,
-                         max_attempts: int = WAIT_MAX_ATTEMPTS,
-                         callback: Optional[typing.Callable[[str], None]] = None) -> str:
+    async def wait_asset(
+            self,
+            asset: dict,
+            delay: int = WAIT_DELAY,
+            max_attempts: int = WAIT_MAX_ATTEMPTS,
+            callback: Optional[Callable[[str],
+                                        None]] = None) -> Dict[Any, Any]:
         """Wait for an item asset to be active.
 
         Prior to waiting for the asset to be active, be sure to activate the
@@ -638,7 +638,7 @@ class DataClient:
                              filename: Optional[str] = None,
                              directory: Path = Path('.'),
                              overwrite: bool = False,
-                             progress_bar: bool = True) -> str:
+                             progress_bar: bool = True) -> Path:
         """Download an asset.
 
         The asset must be active before it can be downloaded. This can be

--- a/planet/clients/data.py
+++ b/planet/clients/data.py
@@ -23,6 +23,7 @@ from .. import exceptions
 from ..constants import PLANET_BASE_URL
 from ..http import Session
 from ..models import Paged, Request, Response, StreamingBody
+from typing import Optional
 
 BASE_URL = f'{PLANET_BASE_URL}/data/v1/'
 SEARCHES_PATH = '/searches'
@@ -74,7 +75,7 @@ class DataClient:
         ```
     """
 
-    def __init__(self, session: Session, base_url: str = None):
+    def __init__(self, session: Session, base_url: Optional[str] = None):
         """
         Parameters:
             session: Open session connected to server.
@@ -107,8 +108,8 @@ class DataClient:
     async def search(self,
                      item_types: typing.List[str],
                      search_filter: dict,
-                     name: str = None,
-                     sort: str = None,
+                     name: Optional[str] = None,
+                     sort: Optional[str] = None,
                      limit: int = 100) -> typing.AsyncIterator[dict]:
         """Execute a quick search.
 
@@ -569,7 +570,7 @@ class DataClient:
                          asset: dict,
                          delay: int = WAIT_DELAY,
                          max_attempts: int = WAIT_MAX_ATTEMPTS,
-                         callback: typing.Callable[[str], None] = None) -> str:
+                         callback: Optional[typing.Callable[[str], None]] = None) -> str:
         """Wait for an item asset to be active.
 
         Prior to waiting for the asset to be active, be sure to activate the
@@ -634,7 +635,7 @@ class DataClient:
 
     async def download_asset(self,
                              asset: dict,
-                             filename: str = None,
+                             filename: Optional[str] = None,
                              directory: Path = Path('.'),
                              overwrite: bool = False,
                              progress_bar: bool = True) -> str:

--- a/planet/clients/orders.py
+++ b/planet/clients/orders.py
@@ -16,7 +16,7 @@
 import asyncio
 import logging
 import time
-import typing
+from typing import AsyncIterator, Callable, List, Optional
 import uuid
 import json
 import hashlib
@@ -26,7 +26,7 @@ from .. import exceptions
 from ..constants import PLANET_BASE_URL
 from ..http import Session
 from ..models import Paged, Request, Response, StreamingBody
-from typing import Optional
+
 
 BASE_URL = f'{PLANET_BASE_URL}/compute/ops'
 STATS_PATH = '/stats/orders/v2'
@@ -202,7 +202,9 @@ class OrdersClient:
         resp = await self._do_request(req)
         return resp.json()
 
-    async def cancel_orders(self, order_ids: Optional[typing.List[str]] = None) -> dict:
+    async def cancel_orders(self,
+                            order_ids: Optional[List[str]] = None
+                            ) -> dict:
         '''Cancel queued orders in bulk.
 
         Parameters:
@@ -275,12 +277,13 @@ class OrdersClient:
                              progress_bar=progress_bar)
         return dl_path
 
-    async def download_order(self,
-                             order_id: str,
-                             directory: Path = Path('.'),
-                             overwrite: bool = False,
-                             progress_bar: bool = False,
-                             checksum: Optional[str] = None) -> typing.List[Path]:
+    async def download_order(
+            self,
+            order_id: str,
+            directory: Path = Path('.'),
+            overwrite: bool = False,
+            progress_bar: bool = False,
+            checksum: Optional[str] = None) -> List[Path]:
         """Download all assets in an order.
 
         Parameters:
@@ -392,12 +395,13 @@ class OrdersClient:
                 raise exceptions.ClientError(
                     f'File ({filename}) checksums do not match.')
 
-    async def wait(self,
-                   order_id: str,
-                   state: Optional[str] = None,
-                   delay: int = 5,
-                   max_attempts: int = 200,
-                   callback: Optional[typing.Callable[[str], None]] = None) -> str:
+    async def wait(
+            self,
+            order_id: str,
+            state: Optional[str] = None,
+            delay: int = 5,
+            max_attempts: int = 200,
+            callback: Optional[Callable[[str], None]] = None) -> str:
         """Wait until order reaches desired state.
 
         Returns the state of the order on the last poll.
@@ -479,7 +483,7 @@ class OrdersClient:
 
     async def list_orders(self,
                           state: Optional[str] = None,
-                          limit: int = 100) -> typing.AsyncIterator[dict]:
+                          limit: int = 100) -> AsyncIterator[dict]:
         """Get all order requests.
 
         Parameters:

--- a/planet/clients/orders.py
+++ b/planet/clients/orders.py
@@ -27,7 +27,6 @@ from ..constants import PLANET_BASE_URL
 from ..http import Session
 from ..models import Paged, Request, Response, StreamingBody
 
-
 BASE_URL = f'{PLANET_BASE_URL}/compute/ops'
 STATS_PATH = '/stats/orders/v2'
 ORDERS_PATH = '/orders/v2'
@@ -203,8 +202,7 @@ class OrdersClient:
         return resp.json()
 
     async def cancel_orders(self,
-                            order_ids: Optional[List[str]] = None
-                            ) -> dict:
+                            order_ids: Optional[List[str]] = None) -> dict:
         '''Cancel queued orders in bulk.
 
         Parameters:
@@ -277,13 +275,12 @@ class OrdersClient:
                              progress_bar=progress_bar)
         return dl_path
 
-    async def download_order(
-            self,
-            order_id: str,
-            directory: Path = Path('.'),
-            overwrite: bool = False,
-            progress_bar: bool = False,
-            checksum: Optional[str] = None) -> List[Path]:
+    async def download_order(self,
+                             order_id: str,
+                             directory: Path = Path('.'),
+                             overwrite: bool = False,
+                             progress_bar: bool = False,
+                             checksum: Optional[str] = None) -> List[Path]:
         """Download all assets in an order.
 
         Parameters:
@@ -395,13 +392,12 @@ class OrdersClient:
                 raise exceptions.ClientError(
                     f'File ({filename}) checksums do not match.')
 
-    async def wait(
-            self,
-            order_id: str,
-            state: Optional[str] = None,
-            delay: int = 5,
-            max_attempts: int = 200,
-            callback: Optional[Callable[[str], None]] = None) -> str:
+    async def wait(self,
+                   order_id: str,
+                   state: Optional[str] = None,
+                   delay: int = 5,
+                   max_attempts: int = 200,
+                   callback: Optional[Callable[[str], None]] = None) -> str:
         """Wait until order reaches desired state.
 
         Returns the state of the order on the last poll.

--- a/planet/clients/orders.py
+++ b/planet/clients/orders.py
@@ -26,6 +26,7 @@ from .. import exceptions
 from ..constants import PLANET_BASE_URL
 from ..http import Session
 from ..models import Paged, Request, Response, StreamingBody
+from typing import Optional
 
 BASE_URL = f'{PLANET_BASE_URL}/compute/ops'
 STATS_PATH = '/stats/orders/v2'
@@ -84,7 +85,7 @@ class OrdersClient:
         ```
     """
 
-    def __init__(self, session: Session, base_url: str = None):
+    def __init__(self, session: Session, base_url: Optional[str] = None):
         """
         Parameters:
             session: Open session connected to server.
@@ -201,7 +202,7 @@ class OrdersClient:
         resp = await self._do_request(req)
         return resp.json()
 
-    async def cancel_orders(self, order_ids: typing.List[str] = None) -> dict:
+    async def cancel_orders(self, order_ids: Optional[typing.List[str]] = None) -> dict:
         '''Cancel queued orders in bulk.
 
         Parameters:
@@ -243,7 +244,7 @@ class OrdersClient:
 
     async def download_asset(self,
                              location: str,
-                             filename: str = None,
+                             filename: Optional[str] = None,
                              directory: Path = Path('.'),
                              overwrite: bool = False,
                              progress_bar: bool = True) -> Path:
@@ -279,7 +280,7 @@ class OrdersClient:
                              directory: Path = Path('.'),
                              overwrite: bool = False,
                              progress_bar: bool = False,
-                             checksum: str = None) -> typing.List[Path]:
+                             checksum: Optional[str] = None) -> typing.List[Path]:
         """Download all assets in an order.
 
         Parameters:
@@ -393,10 +394,10 @@ class OrdersClient:
 
     async def wait(self,
                    order_id: str,
-                   state: str = None,
+                   state: Optional[str] = None,
                    delay: int = 5,
                    max_attempts: int = 200,
-                   callback: typing.Callable[[str], None] = None) -> str:
+                   callback: Optional[typing.Callable[[str], None]] = None) -> str:
         """Wait until order reaches desired state.
 
         Returns the state of the order on the last poll.
@@ -477,7 +478,7 @@ class OrdersClient:
         return current_state
 
     async def list_orders(self,
-                          state: str = None,
+                          state: Optional[str] = None,
                           limit: int = 100) -> typing.AsyncIterator[dict]:
         """Get all order requests.
 

--- a/planet/data_filter.py
+++ b/planet/data_filter.py
@@ -14,7 +14,7 @@
 """Functionality for preparing a data search filter"""
 from datetime import datetime
 import logging
-from typing import Any, Callable, List, Union
+from typing import Optional, Any, Callable, List, Union
 
 from planet import exceptions, geojson
 
@@ -76,7 +76,7 @@ def _range_filter(
     lt: Any,
     gte: Any,
     lte: Any,
-    callback: Callable = None,
+    callback: Optional[Callable] = None,
 ) -> dict:
     """Base for creating range filters.
 
@@ -109,10 +109,10 @@ def _range_filter(
 
 
 def date_range_filter(field_name: str,
-                      gt: datetime = None,
-                      lt: datetime = None,
-                      gte: datetime = None,
-                      lte: datetime = None) -> dict:
+                      gt: Optional[datetime] = None,
+                      lt: Optional[datetime] = None,
+                      gte: Optional[datetime] = None,
+                      lte: Optional[datetime] = None) -> dict:
     """Create a DateRangeFilter
 
     The DateRangeFilter can be used to search on any property with a timestamp
@@ -151,10 +151,10 @@ def _datetime_to_rfc3339(value: datetime) -> str:
 
 
 def range_filter(field_name: str,
-                 gt: float = None,
-                 lt: float = None,
-                 gte: float = None,
-                 lte: float = None) -> dict:
+                 gt: Optional[float] = None,
+                 lt: Optional[float] = None,
+                 gte: Optional[float] = None,
+                 lte: Optional[float] = None) -> dict:
     """Create a RangeFilter
 
     The RangeFilter can be used to search for items with numerical properties.
@@ -179,10 +179,10 @@ def range_filter(field_name: str,
 
 
 def update_filter(field_name: str,
-                  gt: float = None,
-                  lt: float = None,
-                  gte: float = None,
-                  lte: float = None) -> dict:
+                  gt: Optional[float] = None,
+                  lt: Optional[float] = None,
+                  gte: Optional[float] = None,
+                  lte: Optional[float] = None) -> dict:
     """Create an UpdateFilter
 
     The UpdateFilter can be used to filter items by changes to a specified

--- a/planet/http.py
+++ b/planet/http.py
@@ -26,6 +26,7 @@ import httpx
 from .auth import Auth, AuthType
 from . import exceptions, models
 from .__version__ import __version__
+from typing import Optional
 
 # NOTE: configuration of the session was performed using the data API quick
 # search endpoint. These values can be re-tested, tested with a new endpoint or
@@ -225,7 +226,7 @@ class Session(BaseSession):
     ```
     '''
 
-    def __init__(self, auth: AuthType = None):
+    def __init__(self, auth: Optional[AuthType] = None):
         """Initialize a Session.
 
         Parameters:

--- a/planet/order_request.py
+++ b/planet/order_request.py
@@ -15,7 +15,7 @@
 """Functionality for preparing order details for use in creating an order"""
 from __future__ import annotations  # https://stackoverflow.com/a/33533514
 import logging
-from typing import Any, Dict, List
+from typing import Optional, Any, Dict, List
 
 from . import geojson, specs
 
@@ -25,11 +25,11 @@ LOGGER = logging.getLogger(__name__)
 def build_request(name: str,
                   products: List[dict],
                   subscription_id: int = 0,
-                  delivery: dict = None,
-                  notifications: dict = None,
-                  order_type: str = None,
-                  tools: List[dict] = None,
-                  stac: dict = None) -> dict:
+                  delivery: Optional[dict] = None,
+                  notifications: Optional[dict] = None,
+                  order_type: Optional[str] = None,
+                  tools: Optional[List[dict]] = None,
+                  stac: Optional[dict] = None) -> dict:
     '''Prepare an order request.
 
     ```python
@@ -96,7 +96,7 @@ def build_request(name: str,
 def product(item_ids: List[str],
             product_bundle: str,
             item_type: str,
-            fallback_bundle: str = None) -> dict:
+            fallback_bundle: Optional[str] = None) -> dict:
     '''Product description for an order detail.
 
     Parameters:
@@ -131,9 +131,9 @@ def product(item_ids: List[str],
     return product_dict
 
 
-def notifications(email: bool = None,
-                  webhook_url: str = None,
-                  webhook_per_order: bool = None) -> dict:
+def notifications(email: Optional[bool] = None,
+                  webhook_url: Optional[str] = None,
+                  webhook_per_order: Optional[bool] = None) -> dict:
     '''Notifications description for an order detail.
 
     Parameters:
@@ -145,10 +145,10 @@ def notifications(email: bool = None,
     return dict((k, v) for k, v in locals().items() if v)
 
 
-def delivery(archive_type: str = None,
+def delivery(archive_type: Optional[str] = None,
              single_archive: bool = False,
-             archive_filename: str = None,
-             cloud_config: dict = None) -> dict:
+             archive_filename: Optional[str] = None,
+             cloud_config: Optional[dict] = None) -> dict:
     '''Order delivery configuration.
 
     Example:
@@ -196,7 +196,7 @@ def amazon_s3(aws_access_key_id: str,
               aws_secret_access_key: str,
               bucket: str,
               aws_region: str,
-              path_prefix: str = None) -> dict:
+              path_prefix: Optional[str] = None) -> dict:
     '''Amazon S3 Cloud configuration.
 
     Parameters:
@@ -224,8 +224,8 @@ def amazon_s3(aws_access_key_id: str,
 def azure_blob_storage(account: str,
                        container: str,
                        sas_token: str,
-                       storage_endpoint_suffix: str = None,
-                       path_prefix: str = None) -> dict:
+                       storage_endpoint_suffix: Optional[str] = None,
+                       path_prefix: Optional[str] = None) -> dict:
     '''Azure Blob Storage configuration.
 
     Parameters:
@@ -255,7 +255,7 @@ def azure_blob_storage(account: str,
 
 def google_cloud_storage(bucket: str,
                          credentials: str,
-                         path_prefix: str = None) -> dict:
+                         path_prefix: Optional[str] = None) -> dict:
     '''Google Cloud Storage configuration.
 
     Parameters:
@@ -368,8 +368,8 @@ def file_format_tool(file_format: str) -> dict:
 
 
 def reproject_tool(projection: str,
-                   resolution: float = None,
-                   kernel: str = None) -> dict:
+                   resolution: Optional[float] = None,
+                   kernel: Optional[str] = None) -> dict:
     '''Create the API spec representation of a reproject tool.
 
     Parameters:
@@ -389,11 +389,11 @@ def reproject_tool(projection: str,
 
 
 def tile_tool(tile_size: int,
-              origin_x: float = None,
-              origin_y: float = None,
-              pixel_size: float = None,
-              name_template: str = None,
-              conformal_x_scaling: bool = None) -> dict:
+              origin_x: Optional[float] = None,
+              origin_y: Optional[float] = None,
+              pixel_size: Optional[float] = None,
+              name_template: Optional[str] = None,
+              conformal_x_scaling: Optional[bool] = None) -> dict:
     '''Create the API spec representation of a reproject tool.
 
     Parameters:
@@ -414,7 +414,7 @@ def tile_tool(tile_size: int,
     return _tool('tile', parameters)
 
 
-def toar_tool(scale_factor: int = None, ) -> dict:
+def toar_tool(scale_factor: Optional[int] = None, ) -> dict:
     '''Create the API spec representation of a TOAR tool.
 
     Parameters:

--- a/planet/reporting.py
+++ b/planet/reporting.py
@@ -14,9 +14,10 @@
 # the License.
 """Functionality for reporting progress."""
 import logging
+from typing import Optional
 
 from tqdm.asyncio import tqdm
-from typing import Optional
+
 
 LOGGER = logging.getLogger(__name__)
 
@@ -89,7 +90,9 @@ class StateBar(ProgressBar):
         """Simple function to be used as a callback for state reporting"""
         self.update(state=state)
 
-    def update(self, state: Optional[str] = None, order_id: Optional[str] = None):
+    def update(self,
+               state: Optional[str] = None,
+               order_id: Optional[str] = None):
         if state:
             self.state = state
             if self.bar is not None:

--- a/planet/reporting.py
+++ b/planet/reporting.py
@@ -18,7 +18,6 @@ from typing import Optional
 
 from tqdm.asyncio import tqdm
 
-
 LOGGER = logging.getLogger(__name__)
 
 

--- a/planet/reporting.py
+++ b/planet/reporting.py
@@ -16,6 +16,7 @@
 import logging
 
 from tqdm.asyncio import tqdm
+from typing import Optional
 
 LOGGER = logging.getLogger(__name__)
 
@@ -57,8 +58,8 @@ class StateBar(ProgressBar):
 
     def __init__(
         self,
-        order_id: str = None,
-        state: str = None,
+        order_id: Optional[str] = None,
+        state: Optional[str] = None,
         disable: bool = False,
     ):
         """Initialize the object.
@@ -88,7 +89,7 @@ class StateBar(ProgressBar):
         """Simple function to be used as a callback for state reporting"""
         self.update(state=state)
 
-    def update(self, state: str = None, order_id: str = None):
+    def update(self, state: Optional[str] = None, order_id: Optional[str] = None):
         if state:
             self.state = state
             if self.bar is not None:

